### ISSUE TITLE
go get fails on Go < 1.6

### ISSUE THREAD

**Issue by ndarilek** [spf13/hugo#2124]
Just cut-and-pasted the `go get` command from the README. It seems to download the packages successfully, but I don't have a `hugo` command in my GOPATH.

Thanks.

**Comment by bep** [comment]
Please use http://discuss.gohugo.io/ for questions (this may feel like an bug, but is more related to your Go setup).

**Comment by bep** [comment]
I'll reopen this, as this is also happen on our build server:

https://travis-ci.org/spf13/hugo/jobs/128242105

But only on Go 1.5.4. Mystery, hopefully only temporary.

**Comment by ndarilek** [comment]
Glad it got reopened, pretty sure my Go setup is correct so was 
disappointed to have it just closed without at least a request for more 
clarification. :)

FWIW, I'm on 1.5.4 as well, as that's the latest I get with Fedora 23. I 
downloaded a prebuilt release so this isn't a blocker, but thought you'd 
appreciate the heads-up that `go get` wasn't working for some of us.

Thanks.

**Comment by bep** [comment]
Sure, I will investigate further if this continues.

**Comment by bep** [comment]
Seems we're not alone; he/she deleted the master branch; looks like Go 1.6 handles it fine, Go 1.5 not so much. Looks like we should look into vendoring sooner rather than later ...

**Comment by mohae** [comment]
In looking into this, I encounter the problem with 1.6.2.  On a test machine I updated Go to 1.6.2 and did a `go get` for hugo.  The output was:

```
go get -u -v github.com/spf13/hugo
github.com/spf13/hugo (download)
github.com/fsnotify/fsnotify (download)
Fetching https://golang.org/x/sys/unix?go-get=1
Parsing meta tags from https://golang.org/x/sys/unix?go-get=1 (status code 200)
get "golang.org/x/sys/unix": found meta tag main.metaImport{Prefix:"golang.org/x/sys", VCS:"git", RepoRoot:"https://go.googlesource.com/sys"} at https://golang.org/x/sys/unix?go-get=1
get "golang.org/x/sys/unix": verifying non-authoritative meta tag
Fetching https://golang.org/x/sys?go-get=1
Parsing meta tags from https://golang.org/x/sys?go-get=1 (status code 200)
golang.org/x/sys (download)
github.com/kardianos/osext (download)
github.com/spf13/afero (download)
github.com/pkg/sftp (download)
github.com/kr/fs (download)
Fetching https://golang.org/x/crypto/ssh?go-get=1
Parsing meta tags from https://golang.org/x/crypto/ssh?go-get=1 (status code 200)
get "golang.org/x/crypto/ssh": found meta tag main.metaImport{Prefix:"golang.org/x/crypto", VCS:"git", RepoRoot:"https://go.googlesource.com/crypto"} at https://golang.org/x/crypto/ssh?go-get=1
get "golang.org/x/crypto/ssh": verifying non-authoritative meta tag
Fetching https://golang.org/x/crypto?go-get=1
Parsing meta tags from https://golang.org/x/crypto?go-get=1 (status code 200)
golang.org/x/crypto (download)
Fetching https://golang.org/x/crypto/curve25519?go-get=1
Parsing meta tags from https://golang.org/x/crypto/curve25519?go-get=1 (status code 200)
get "golang.org/x/crypto/curve25519": found meta tag main.metaImport{Prefix:"golang.org/x/crypto", VCS:"git", RepoRoot:"https://go.googlesource.com/crypto"} at https://golang.org/x/crypto/curve25519?go-get=1
get "golang.org/x/crypto/curve25519": verifying non-authoritative meta tag
Fetching https://golang.org/x/text/transform?go-get=1
Parsing meta tags from https://golang.org/x/text/transform?go-get=1 (status code 200)
get "golang.org/x/text/transform": found meta tag main.metaImport{Prefix:"golang.org/x/text", VCS:"git", RepoRoot:"https://go.googlesource.com/text"} at https://golang.org/x/text/transform?go-get=1
get "golang.org/x/text/transform": verifying non-authoritative meta tag
Fetching https://golang.org/x/text?go-get=1
Parsing meta tags from https://golang.org/x/text?go-get=1 (status code 200)
golang.org/x/text (download)
Fetching https://golang.org/x/text/unicode/norm?go-get=1
Parsing meta tags from https://golang.org/x/text/unicode/norm?go-get=1 (status code 200)
get "golang.org/x/text/unicode/norm": found meta tag main.metaImport{Prefix:"golang.org/x/text", VCS:"git", RepoRoot:"https://go.googlesource.com/text"} at https://golang.org/x/text/unicode/norm?go-get=1
get "golang.org/x/text/unicode/norm": verifying non-authoritative meta tag
github.com/spf13/cast (download)
github.com/spf13/jwalterweatherman (download)
github.com/spf13/cobra (download)
github.com/spf13/pflag (download)
github.com/cpuguy83/go-md2man (download)
github.com/russross/blackfriday (download)
github.com/shurcooL/sanitized_anchor_name (download)
github.com/spf13/fsync (download)
github.com/PuerkitoBio/purell (download)
github.com/opennota/urlesc (download)
github.com/kyokomi/emoji (download)
github.com/miekg/mmark (download)
github.com/BurntSushi/toml (download)
github.com/mitchellh/mapstructure (download)
github.com/spf13/viper (download)
github.com/hashicorp/hcl (download)
github.com/kr/pretty (download)
github.com/kr/text (download)
github.com/magiconair/properties (download)
# cd /home/vagrant/code/go/src/github.com/magiconair/properties; git pull --ff-only
fatal: Not possible to fast-forward, aborting.
package github.com/magiconair/properties: exit status 128
Fetching https://gopkg.in/yaml.v2?go-get=1
Parsing meta tags from https://gopkg.in/yaml.v2?go-get=1 (status code 200)
get "gopkg.in/yaml.v2": found meta tag main.metaImport{Prefix:"gopkg.in/yaml.v2", VCS:"git", RepoRoot:"https://gopkg.in/yaml.v2"} at https://gopkg.in/yaml.v2?go-get=1
gopkg.in/yaml.v2 (download)
github.com/bep/inflect (download)
github.com/dchest/cssmin (download)
github.com/eknkc/amber (download)
github.com/yosssi/ace (download)
github.com/spf13/nitro (download)
github.com/gorilla/websocket (download)
```

In looking at $GOPATH/src/github.com/kr/text the only branch it has is master, so I'm not sure what is going on at this point.

**Comment by bep** [comment]
>In looking at $GOPATH/src/github.com/kr/text the only branch it has is master, so I'm not sure what is going on at this point. 

What you are looking at is unrelated (you have repo out of synch -- aka someone did a push -f ... annoying but happens).

The issue described here doesn't happen in Go 1.6.*.

**Comment by mohae** [comment]
Oops, I pasted the wrong output.  Also when I redid it on a new VM it worked as expected...sorry about the noise.

**Comment by mohae** [comment]
This issue does raise the question of what Hugo's policy for support of non-current versions of Go is, or should be.  Sorry it that's been stated elsewhere and I missed it.

The installation instructions for Go always point people to the latest release of Go.  In my experience, it is a common expectation that developers will be on the current release of Go, unless they have specific needs or reasons not to be.  One example would be JuJu and sticking with 1.4 (the last I checked) because of the longer compile times introduced by switching to the pure Go compiler. 

I realize that there are people that prefer using their distro's package manager to install Go but that often results in an old version of Go and the Go installation instructions does not involve using package managers: https://golang.org/doc/install.  In my experience Go installed by package managers are usually outdated and may be missing executables that are normally part of Go, e.g. .`godoc` and `gofmt`.  

As an example, I installed Go, using the package manager on a couple of Ubuntu LTS boxes.  Doing a `sudo apt-get install golang-go` on Ubuntu 14.04 results in `go1` being installed which .  This is an extreme example due to the age of 14.04.  Doing the same thing on 16.04 results in `go1.6.1 linux/amd64`  since 16.04 was just released.  However, the drift will get worse as time progresses.

While Go does have a backwards compatibility guarantee, there are exceptions and it does not cover everything.  This is a situation where the library in question modified their repo to take advantage of new functionality and it does not appear that the owner of the package will revert the change so that it will work with < 1.6.  I may be wrong, but @kr's comment on https://github.com/golang/gddo/issues/405 leads me to this conclusion.
